gcloud/storage/_implicit_environ.py

```diff
@@ -81,29 +29,11 @@ class _DefaultsContainer(object):
     :param connection: Persistent implied connection from environment.
     """
 
-    @_lazy_property_deco
-    @staticmethod
-    def project():
-        """Return the implicit default project."""
-        return _determine_default_project()
-
-    def __init__(self, project=None, bucket=None, connection=None,
-                 implicit=False):
-        if project is not None or not implicit:
-            self.project = project
+    def __init__(self, bucket=None, connection=None):
```

Comment:
I figured out later that we have to have somewhere to store package-specific defaults (like bucket, plus the locally-specified connection).  The class is so simple that it makes no sense to do anything but replicate it.